account_invoice_production_lot/report/report_invoice.xml

```diff
@@ -2,11 +2,26 @@
 <odoo>
 
     <template id="account_invoice_production_lot_report" inherit_id="account.report_invoice_document">
-        <span t-field="line.name" position="after">
-            <t t-if="line[:1].lot_formatted_note">
-                <div style="margin-left:25px;" t-field="line[:1].lot_formatted_note"/>
+        <xpath expr="//t[@name='account_invoice_line_accountable']/.." position="after">
+            <t t-if="not line.display_type and line.product_id.tracking != 'none'" name="account_invoice_line_lot">
```

Comment:
In the case of unique serial numbers, displaying one for each line can result in very long reports. In that case, displaying them altogether separated by commas would be preferable as proposed here: https://github.com/OCA/account-invoice-reporting/pull/130